src/layer/vector/Canvas.js

```diff
@@ -135,9 +135,28 @@ export var Canvas = Renderer.extend({
 		}
 	},
 
+	_onKeyPress: function (layer, e) {
+		var data = {
+			originalEvent: e
+		};
+		layer.fire('keypress', data);
+	},
+
 	_initPath: function (layer) {
 		this._updateDashArray(layer);
 		this._layers[Util.stamp(layer)] = layer;
+		if (layer.options.keyboard) {
+			layer._path = DomUtil.create('div', 'leaflet-canvas-interactive', this._container);
+			layer._path.setAttribute('tabindex', 0);
+			layer._path.setAttribute('role', 'graphics-symbol img');
+			layer._path.setAttribute('aria-label', layer.options.title);
+			layer._path.setAttribute('title', layer.options.description);
```

Comment:
I wonder if we should go with:

```diff
-layer._path.setAttribute('aria-label', layer.options.title);
+layer._path.setAttribute('title', layer.options.title);

-layer._path.setAttribute('title', layer.options.description);
+layer._path.setAttribute('aria-description', layer.options.description);
```

[`aria-description`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-description) is a relatively new ARIA attribute, and [not yet supported by VoiceOver](https://a11ysupport.io/tech/aria/aria-description_attribute#support-table-0). But it may be the correct thing to use, at least in the long run. 🤔

Reply:
I am testing the changes using VoiceOver on a mac laptop with 3 browsers: Firefox, Chrome & Safari. `aria-description` is announced in all of them.

[The link you mentioned](https://a11ysupport.io/tech/aria/aria-description_attribute#support-table-0) says it does not work for the `input` tag, but we use `div`, maybe that's the reason.

Reply:
That's good to hear!

Yes, a11ysupport.io doesn't always test the most simple cases, which is kind of unfortunate. But that initiative was created and mostly maintained by one individual.

Reply:
Oh, wait, I misunderstood you. `aria-description` does not work for me with VoiceOver. I was talking about the current implementation with `title` and `aria-label`.

If I use `aria-description` it is **not** announced as an additional description. But `title` is announced as an element name.
If I use `aria-label` & `title`, then `aria-label` is announced as an element name and `title` is used as a description.

Reply:
Ah, unfortunate. With the current approach there will be confusion with `title` option setting `aria-label` and `desc(ription)` setting `title` attribute.

Reply:
> Ah, unfortunate. With the current approach it may confuse people that the title option is setting aria-label and description option is setting the title attribute.

Agree, sounds confusing. We can introduce better names for those options. The name might be more independent of implementation. Something like `accessibleName` & accessibleDescription`. What do you think?

Reply:
I've thought of that (well `accName` and `accDesc`), but we'd still be stuck with `alt` and `title` for markers to avoid breakage, so I think we'd want `title` here as well.

Reply:
To sum up, we have thee options:
- use `title` and `desc` as option names. Use `title` & `aria-description` in canvas sub DOM. 
The naming is more consistent with the `Marker`, but this does not currently work with VoiceOver.
- use `title` & `desc` as option names. Use `aria-label` & `title` in canvas sub DOM.
The naming is more consistent with the `Marker`, but we introduce a source of confusion since `option.title` does not go into the `title` attribute.
- use `accName` & `accDesc` as option names. Use `aria-label` & `title` in canvas sub DOM.
The naming is not consistent with the `Marker`, and I do not believe we will change the option names of Marker. But the Voice Over works and works the same for canvas & SVG renderer.

I prefer the last option and am curious what others think.

Reply:
> - use `accName` & `accDesc` as option names. [...]
>    The naming is not consistent with the `Marker`, and I do not believe we will change the option names of Marker.

With this option, I suppose there's also the possibility to introduce `accName` and `accDesc` as synonyms for `Marker`'s `alt` and `title` for consistency? I think consistency is appreciated by developers.

However I'm kind of leaning towards keeping just `alt` and `title` for Marker, and `desc` and `title` for canvas and SVG. I'm not opposed to use `aria-description` for the `desc` option, with a note in the docs stating it is not fully supported (with a link to a11ysupport.io) which could simply be removed in the future. Assuming this is an acceptable approach:
- developers will always know that the `title` option will produce the browser's native tooltips
- there's consistency
- the `title` options always give exactly that: `title` attribute, or `<title>`

Reply:
Other options include:

- `aria-describedby` instead of `aria-label` (and use for the `desc` option) since `aria-description` isn't fully supported
- only implement `title` for now

Reply:
> Do you think we should add another element to this `<div>`?

Possibly. Or sibling, but I don't think that should be necessary. The following test HTML works (tested in NVDA, please test in VO):

```html
<canvas>
  <div tabindex="0" role="graphics-symbol img" title="title" aria-describedby="leaflet-canvas-interactive-desc">
    <span id="leaflet-canvas-interactive-desc">desc</span>
  </div>
</canvas>
```

In this case `aria-describedby` is used only because the roles do not support [name from content](https://www.w3.org/TR/wai-aria-1.1/#namefromcontent).

This would result in better consistency in the names, but also allow us to change `aria-describedby` to `aria-description` (to have only 1 element) in the future.

Reply:
I like this approach and have implemented it. 
Tested it with VoiceOver on Safari, Firefox and Chrome. 
It works consistently on all browsers and narrates SVG and canvas versions in the same way.

Reply:
Awesome, great work! 🙂 Canvas sub DOM is pretty cool.